wordpress-mobile/WordPress-iOS
# Makes the domain searches immediate.

Fixes #17240 

Notes:

- Live domain searches are debounced with a minimum delay of 0.5 secs.
- The search-in-progress HUD is now shown on top of the results only.  It looked kind of odd when the device keyboard was shown.

## Associated PRs:

WordPress Shared: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/300

## To test:

1. Go into the Domain Registration flow.
2. In the Domain Suggestions screen start typing.

Make sure searches are triggered 0.5 seconds after the user stops typing.  Make sure the whole search feature works as expected.

## Regression Notes

1. Potential unintended areas of impact

I've slightly modified `Debouncer.swift` which is used across the app for different purposes.

Outside of that, the edited components are only used for the new Domain Registration flow, so they should not affect existing behaviour.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

This class has existing unit tests to verify it's previous behaviour, and those tests are still succeeding.

3. What automated tests I added (or what prevented me from doing so)

I've added a test to document and verify the new behaviour that I've added to this class.

PR submission checklist:
